### ISSUE TITLE
prost 0.3 update breaks Conduit generated code

### ISSUE THREAD

**Issue by hawkw** [tower-rs/tower-grpc#45]
After updating [my Conduit branch that depends on `tower-grpc` from this repo](https://github.com/runconduit/conduit/pull/225) to depend on `prost` 0.3.0, I see the following compile error:

```
   Compiling conduit-proxy v0.1.3 (file:///Users/eliza/Code/go/src/github.com/runconduit/conduit/proxy)
error[E0432]: unresolved import `super::Update`
   --> /Users/eliza/Code/go/src/github.com/runconduit/conduit/target/debug/build/conduit-proxy-e25c6e4edf2849f0/out/conduit.proxy.telemetry.rs:147:17
    |
147 |     use super::{Update, ReportRequest, ReportResponse};
    |                 ^^^^^^ no `Update` in `control::pb::proxy::telemetry`

error: aborting due to previous error

error: Could not compile `conduit-proxy`.

```

**Comment by hawkw** [comment]
Interesting, it looks like this is happening because the code generator is now outputting all the client code for `Destination` into the `telemetry::client` module, as well as the the generated code for `Telemetry`:

conduit.proxy.telemetry.rs:
```
#[derive(Clone, PartialEq, Message)]
pub struct ReportRequest {
    #[prost(message, optional, tag="1")]
    pub process: ::std::option::Option<Process>,
    #[prost(enumeration="report_request::Proxy", tag="2")]
    pub proxy: i32,
    #[prost(message, repeated, tag="3")]
    pub server_transports: ::std::vec::Vec<ServerTransport>,
    #[prost(message, repeated, tag="4")]
    pub client_transports: ::std::vec::Vec<ClientTransport>,
    #[prost(message, repeated, tag="5")]
    pub requests: ::std::vec::Vec<RequestScope>,
}
pub mod report_request {
    #[derive(Clone, Copy, Debug, PartialEq, Eq, Enumeration)]
    pub enum Proxy {
        Inbound = 0,
        Outbound = 1,
    }
}
#[derive(Clone, PartialEq, Message)]
pub struct Process {
    #[prost(string, tag="1")]
    pub node: String,
    #[prost(string, tag="2")]
    pub scheduled_instance: String,
    #[prost(string, tag="3")]
    pub scheduled_namespace: String,
}
#[derive(Clone, PartialEq, Message)]
pub struct ServerTransport {
    #[prost(message, optional, tag="1")]
    pub source_ip: ::std::option::Option<super::super::common::IpAddress>,
    #[prost(uint32, tag="2")]
    pub connects: u32,
    #[prost(message, repeated, tag="3")]
    pub disconnects: ::std::vec::Vec<TransportSummary>,
    #[prost(enumeration="super::super::common::Protocol", tag="4")]
    pub protocol: i32,
}
#[derive(Clone, PartialEq, Message)]
pub struct ClientTransport {
    #[prost(message, optional, tag="1")]
    pub target_addr: ::std::option::Option<super::super::common::TcpAddress>,
    #[prost(uint32, tag="2")]
    pub connects: u32,
    #[prost(message, repeated, tag="3")]
    pub disconnects: ::std::vec::Vec<TransportSummary>,
    #[prost(enumeration="super::super::common::Protocol", tag="4")]
    pub protocol: i32,
}
#[derive(Clone, PartialEq, Message)]
pub struct TransportSummary {
    #[prost(uint64, tag="1")]
    pub duration_ms: u64,
    #[prost(uint64, tag="2")]
    pub bytes_sent: u64,
}
#[derive(Clone, PartialEq, Message)]
pub struct RequestScope {
    #[prost(message, optional, tag="1")]
    pub ctx: ::std::option::Option<RequestCtx>,
    #[prost(uint32, tag="2")]
    pub count: u32,
    #[prost(message, repeated, tag="3")]
    pub responses: ::std::vec::Vec<ResponseScope>,
}
#[derive(Clone, PartialEq, Message)]
pub struct RequestCtx {
    #[prost(message, optional, tag="1")]
    pub source_ip: ::std::option::Option<super::super::common::IpAddress>,
    #[prost(message, optional, tag="2")]
    pub target_addr: ::std::option::Option<super::super::common::TcpAddress>,
    #[prost(string, tag="3")]
    pub authority: String,
    #[prost(message, optional, tag="4")]
    pub method: ::std::option::Option<super::super::common::HttpMethod>,
    #[prost(string, tag="5")]
    pub path: String,
}
/// A latency value in tenths of a millisecond and a count of the times
/// that latency was observed.
#[derive(Clone, PartialEq, Message)]
pub struct Latency {
    /// Latency value in tenths of a millisecond.
    #[prost(uint32, tag="1")]
    pub latency: u32,
    /// Count of occurences of this latency value.
    #[prost(uint32, tag="2")]
    pub count: u32,
}
#[derive(Clone, PartialEq, Message)]
pub struct ResponseScope {
    #[prost(message, optional, tag="1")]
    pub ctx: ::std::option::Option<ResponseCtx>,
    /// Response latencies (time from request headers sent to response headers received),
    /// represented as an array of observed latency values with precision to 100µs and
    /// the number of times those values were observed, ordered by the latency value.
    #[prost(message, repeated, tag="2")]
    pub response_latencies: ::std::vec::Vec<Latency>,
    #[prost(message, repeated, tag="3")]
    pub ends: ::std::vec::Vec<EosScope>,
}
#[derive(Clone, PartialEq, Message)]
pub struct ResponseCtx {
    #[prost(uint32, tag="1")]
    pub http_status_code: u32,
}
#[derive(Clone, PartialEq, Message)]
pub struct EosScope {
    #[prost(message, optional, tag="1")]
    pub ctx: ::std::option::Option<EosCtx>,
    #[prost(message, repeated, tag="2")]
    pub streams: ::std::vec::Vec<StreamSummary>,
}
#[derive(Clone, PartialEq, Message)]
pub struct EosCtx {
    #[prost(oneof="eos_ctx::End", tags="1, 2, 3")]
    pub end: ::std::option::Option<eos_ctx::End>,
}
pub mod eos_ctx {
    #[derive(Clone, Oneof, PartialEq)]
    pub enum End {
        #[prost(uint32, tag="1")]
        GrpcStatusCode(u32),
        #[prost(uint32, tag="2")]
        ResetErrorCode(u32),
        /// Stream ended without reset and without grpc status code
        #[prost(bool, tag="3")]
        Other(bool),
    }
}
#[derive(Clone, PartialEq, Message)]
pub struct StreamSummary {
    #[prost(uint64, tag="1")]
    pub duration_ms: u64,
    #[prost(uint64, tag="2")]
    pub bytes_sent: u64,
    #[prost(uint32, tag="3")]
    pub frames_sent: u32,
}
#[derive(Clone, PartialEq, Message)]
pub struct ReportResponse {
}
pub mod client {
    use ::tower_grpc::codegen::client::*;
    use super::{Update, ReportRequest, ReportResponse};

    #[derive(Debug)]
    pub struct Destination<T> {
        inner: grpc::Grpc<T>,
    }

    impl<T> Destination<T>
    where T: tower_h2::HttpService,
    {
        pub fn new(inner: T) -> Self {
            let inner = grpc::Grpc::new(inner);
            Self { inner }
        }

        pub fn poll_ready(&mut self) -> futures::Poll<(), grpc::Error<T::Error>> {
            self.inner.poll_ready()
        }

        pub fn get(&mut self, request: grpc::Request<super::super::super::common::Destination>) -> grpc::server_streaming::ResponseFuture<Update, T::Future>
        where grpc::unary::Once<super::super::super::common::Destination>: grpc::Encodable<T::RequestBody>,
        {
            let path = http::PathAndQuery::from_static("/conduit.proxy.destination.Destination/Get");
            self.inner.server_streaming(request, path)
        }
    }

    #[derive(Debug)]
    pub struct Telemetry<T> {
        inner: grpc::Grpc<T>,
    }

    impl<T> Telemetry<T>
    where T: tower_h2::HttpService,
    {
        pub fn new(inner: T) -> Self {
            let inner = grpc::Grpc::new(inner);
            Self { inner }
        }

        pub fn poll_ready(&mut self) -> futures::Poll<(), grpc::Error<T::Error>> {
            self.inner.poll_ready()
        }

        pub fn report(&mut self, request: grpc::Request<ReportRequest>) -> grpc::unary::ResponseFuture<ReportResponse, T::Future, T::ResponseBody>
        where grpc::unary::Once<ReportRequest>: grpc::Encodable<T::RequestBody>,
        {
            let path = http::PathAndQuery::from_static("/conduit.proxy.telemetry.Telemetry/Report");
            self.inner.unary(request, path)
        }
    }
}
```

**Comment by hawkw** [comment]
WELP, I found the problem: https://github.com/tower-rs/tower-grpc/commit/bbf1faca7e6e04aa611117e83d318dae44fc3835#diff-d15ac6177e75577c93c4041346f98ed0L125

**Comment by carllerche** [comment]
Ah, lame. Ideally we would have a test that would catch this. I'm not sure why the existing ones did not.

**Comment by hawkw** [comment]
@carllerche 
> Ideally we would have a test that would catch this. I'm not sure why the existing ones did not.

I tried to build `test/multifile` on master, and it does fail to compile as expected; turns out we just weren't trying to build it on CI. Fixed that in #46! :)
